Feat/mypage database

mintnaka opened the pull request:
- 마이페이지 DB 연결 및 업적 보상 이벤트 구현
- 클래스명, 주석 변경 및 코드 정리

## PR 유형

- [x] 새로운 기능 추가
- [ ] 버그 수정
- [x] CSS 등 사용자 UI 디자인 변경
- [x] 코드에 영향을 주지 않는 변경사항(오타 수정, 탭 사이즈 변경, 변수명 변경)
- [x] 코드 리팩토링
- [x] 주석 추가 및 수정
- [ ] 문서 수정
- [ ] 테스트 추가, 테스트 리팩토링
- [ ] 빌드 부분 혹은 패키지 매니저 수정
- [ ] 파일 혹은 폴더명 수정
- [ ] 파일 혹은 폴더 삭제

## PR 체크리스트

- [x] 커밋 메시지 컨벤션에 맞게 작성했습니다.
- [x] 변경 사항에 대한 테스트를 했습니다.(버그 수정/기능에 대한 테스트).

### PR 상세
https://github.com/user-attachments/assets/e77bc5e8-f4ea-4847-aae1-8833f43ab4ce

![image](https://github.com/user-attachments/assets/b30c2987-6c5b-48c2-9f92-bfe396e14f41)
![image](https://github.com/user-attachments/assets/b28be2e7-ef0a-4ca3-a0eb-11cf13f34856)
![image](https://github.com/user-attachments/assets/ee8f400a-7000-41ac-8311-aafd72e1d8f5)

## 이슈

resolves #72

mintnaka committed:
[Feat] 마이페이지 DB 연결 1

- 프로필 헤더 연결 완료
- 프로필 정보 연결 완료
- 통계 부분 연결 완료
- 날짜 가져오는 함수 utils에 추가
- 통계 업적 카드 데이터 types와 utils 파일로 관리 시도 중인데 업적 부분 DB 연결 때문에 수정 필요

#72

mintnaka committed:
[Modify] 마이페이지 프로필 정보란 수정

- 코드 개선
- 대표 취미 아이콘 렌더링 부분 코드 수정
- 유효성 검사 강화

#72

mintnaka committed:
[Feat] 마이페이지 DB 연결 2
- 업적 기본적인 DB 연결 완료
- 레벨업 이벤트 구현중

#72

mintnaka committed:
Merge branch 'develop' of https://github.com/FRONTENDBOOTCAMP-12th/HobbyDobby-3rd into feat/mypage-database

mintnaka committed:
Merge branch 'develop' of https://github.com/FRONTENDBOOTCAMP-12th/HobbyDobby-3rd into feat/mypage-database

mintnaka committed:
[Feat] 마이페이지 프로필 사진 DB 연결

- 연결 확인
- 프로필 이미지 CSS 조정

#72

mintnaka committed:
Merge branch 'develop' of https://github.com/FRONTENDBOOTCAMP-12th/HobbyDobby-3rd into feat/mypage-database

mintnaka committed:
[Feat] 마이페이지 DB 연결 3

- 업적 보상 이벤트(버튼) 기능 테스트 구현

#72

mintnaka committed:
[Feat/Design] 마이페이지 DB 연결 마지막

- 주석 추가 및 코드 정리(가독성 개선)

#72

mintnaka committed:
Merge branch 'develop' of https://github.com/FRONTENDBOOTCAMP-12th/HobbyDobby-3rd into feat/mypage-database

mintnaka committed:
[Comment] api 파일 주석 수정

#72

mintnaka committed:
[Build] pnpm i로 패키지 업데이트

kongsh reviewed (changes requested):
- StatCardList, AchievementCardList에서 getUserCompletedChallenge() 함수가 반복되는데, 이를 둘을 감싸는 상위 컴포넌트에서 한번만 실행 한 뒤 결과값을 props로 넘겨주세요
- api.ts 파일에 작성하신 함수들 기능에 맞게 정리해주세요.

mintnaka committed:
[Refactor/Bugfix] 마이페이지 DB 수정 요청 반영

- api 파일 코드 정리(기능별)
- 통계와 업적 카드에 동일하게 사용되는 completedChallenges 데이터 패치 함수 상위 컴포넌트로 이동해서 관리
- 리팩토링 과정에서 발견한 버그 고침
  - 업적 현재 값이 0인 경우와 진행도가 꽉 차 레벨업 후 0으로 초기화되어야 하는 경우를 구분하기 위해 조건문 추가

#72

kongsh reviewed (approved):
확인했습니다!

kongsh committed:
Merge branch 'develop' into feat/mypage-database
